# PhaseGrass_review  
```
Response to reviewers' points:

1) HapHiC for Sikem

What I did is the following:

# 03.09.2025
# install HapHiC
git clone https://github.com/zengxiaofei/HapHiC.git
mamba env create -p /scratch/yutang/haphic --file HapHiC/conda_env/environment_py310.yml
# unfortunately, above line returned error:
# ERROR: Could not find a version that satisfies the requirement contourpy==1.2.1 (from versions: none)
# ERROR: No matching distribution found for contourpy==1.2.1

# so now try a different yml in the HapHiC repo
mamba env create -p /scratch/yutang/haphic --file HapHiC/conda_env/environment_py312.yml
# got the same error.

# try manual installation following steps in create_conda_env_py312.sh
mamba create -n haphic python=3.12.4
mamba activate haphic
mamba install mkl
mamba install sparse_dot_mkl "numpy<2.0.0"
pip3 install scipy matplotlib
pip3 install scikit-learn networkx
pip3 install pysam
pip3 install portion
# I found I need set the proxy to use pip for our machine. Previous installation faliure mighe be due to this proxy thing
# install bwa samblaster samtools
mamba install bioconda::bwa bioconda::samblaster bioconda::samtools

# now test HapHiC with the unitig assembly of Sikem generated by Hifiasm
# create a folder for testing with haphic
mkdir haphic_utg

# copy Sikem hifiasm p_utg assembly to this folder
cp hifiasm_ctg/sikem.asm.p_utg.fasta haphic_utg/Sikem_utg.fasta
# this utg assembly was generated while I was doing PhaseGrass with below command
# hifiasm -o sikem.asm -t 60 -l0 -u 0 ../all_hifi.fastq.gz

# copy hic data of Sikem to the folder
mv /scratch/yutang/manual_curation/sikem_dip/fastq/* haphic_utg
# HiC includes the following two, Hic_reads_R1.fastq.gz, Hic_reads_R2.fastq.gz

# (1) Align Hi-C data to the assembly, remove PCR duplicates and filter out secondary and supplementary alignments
bwa index Sikem_utg.fasta
bwa mem -5SP -t 32 Sikem_utg.fasta Hic_reads_R1.fastq.gz Hic_reads_R2.fastq.gz | samblaster | samtools view - -@ 32 -S -h -b -F 3340 -o HiC.bam

# (2) Filter the alignments with MAPQ 1 (mapping quality â‰¥ 1) and NM 3 (edit distance < 3)
/scratch/yutang/HapHiC/utils/filter_bam HiC.bam 1 --nm 3 --threads 32 | samtools view - -b -@ 32 -o HiC.filtered.bam

# 04.09.2025
# (3) run haphic pipeline
/scratch/yutang/HapHiC/haphic pipeline Sikem_utg.fasta HiC.filtered.bam 14 --threads 32 --processes 32
# haphic pipeline finished successfully with 0 error
# this resulted in scaffolds.fa, the final scaffolds 

# generate hic matrix for the scaffolds generated by haphic
ln -s /scratch/yutang/sikem/haphic_utg/Sikem_utg.fasta .
samtools faidx Sikem_utg.fasta
/scratch/yutang/HapHiC/scripts/../utils/juicer pre -a -q 1 -o out_JBAT /scratch/yutang/sikem/haphic_utg/HiC.filtered.bam scaffolds.raw.agp Sikem_utg.fasta.fai >out_JBAT.log 2>&1
(java -Djava.awt.headless=true -jar -Xmx32G /scratch/yutang/HapHiC/scripts/../utils/juicer_tools.1.9.9_jcuda.0.8.jar pre out_JBAT.txt out_JBAT.hic.part <(cat out_JBAT.log | grep PRE_C_SIZE | awk '{print $2" "$3}')) &(mvout_JBAT.hic.part out_JBAT.hic)

```
